### PR TITLE
Improve pylint score of `physics` module

### PR DESCRIPTION
**Pull request by sduquemesa**
**Context:**
Mr Mustard uses `pylint` for code quality checking.

**Description of the Change:**
- This PR refactors the module `physics` so that `pylint mrmustard/physics` code rating goes from `6.65/10` to `9.89/10`. 
- A couple of new docstrings are included.

**Benefits:**
`pylint` score of `physics` module is improved.

### CODE REVIEW

File: mrmustard/physics/__init__.py [hunk 1]
```diff
@@ -87,7 +88,7 @@ def overlap(A, B) -> float:
     Returns:
         float: the overlap between the two states
     """
-    raise NotImplementedError
```
Comment (thisac) [inline mrmustard/physics/__init__.py:90]: How come an exception was raised here? Was this just a mistake/bug? 🤔

File: mrmustard/physics/fock.py [hunk 2]
```diff
@@ -366,10 +371,19 @@ def contract_states(
...
 def normalize(fock: Tensor, is_dm: bool):
+    r"""
+    Returns the normalized ket state.
+
+    Args:
+        fock (Tensor): the state to be normalized
+        is_dm (optioanl bool): whether the input tensor is a density matrix
+    Returns:
+        Tensor: the normalized state
+    """
```
Comment (thisac) [inline mrmustard/physics/fock.py:382]:
```suggestion
    r"""Returns the normalized ket state.

    Args:
        fock (Tensor): the state to be normalized
        is_dm (optioanl bool): whether the input tensor is a density matrix

    Returns:
        Tensor: the normalized state
    """
```

**Review by thisac — APPROVED** [review]
Looks good! Just a few minor comments. 💯

File: mrmustard/physics/__init__.py (hunk 1 above)
Comment (sduquemesa) [reply mrmustard/physics/__init__.py]: I believe this functions were either not correctly implemented or thoroughly tested before release so they were set as not implemented.

I'm leaving them untouched for this PR and will check implementation+tests on a dedicated pull request.

**Pull request merged**